### ISSUE TITLE
Audio/video chat room

### ISSUE THREAD

**Issue by lazerwalker** [lazerwalker/azure-mud#8]
Naive first implementation: 

* Upon entering a special room, you're added to a videochat with everyone in there, using a WebRTC full-mesh topology
* Toggles to not send your video and/or not receive video from anyone else
* Toggle to receive captions
    * If anyone toggles this, every single client generates captions for their own speech, and sends the raw captions over p2p data channel to people who request it. Gets displayed in chat scroll.
* Toggle to do text-to-speech
    * Submits mp3 taken from tts-ing what you type in the text chat
* For v1, no max room size. Good thing to test.

**Comment by lazerwalker** [comment]
The technical underpinnings of spinning up a WebRTC full-mesh topology are in place (just call "broadcastPeerId", and all peers in the same room will attempt to connect to you)

What's needed:

- [ ] When in a room with media chat, add a 'join' button that, when clicked, calls broadcastPeerId
- [ ] Next to the button, show how many people (who?) are in chat
- [ ] If you receive a broadcastPeerId request, don't respond unless you're already in webRTC mode
- [ ] AV selector
- [ ] Toggle to disable sending your video/audio
- [ ] Toggle to disable sending others' video
- [ ] Toggle to receive captions
    - Send message to everyone who's connected that you want captions.
    - When you receive caption request from at least 1 person, STT everything you say. Broadcast that over webrtc data to that person
    - When you receive caption data, display it as a message that indicates it's a transcript
- [ ] Toggle text-to-speech
   - If enabled, when you send a text message, either: generate an mp3 and stream that data, generate an mp3 and data-send the URL, or tell each other client to use client-side TTS

**Comment by lazerwalker** [comment]
Some WIP work was done to let you mute others' streams, but it seems not viable. It's on the `mute-others` branch.

I tried removing tracks from the stream for a given peer, but WebRTC explicitly yelled at me that I couldn't re-add a removed track, I needed to enable/disable the track.

However, right now, there's a single universal stream/track across all peers, so if I enable/disable, it will do it globally for all peers you're sending data to.

Open question: I can `clone()` a MediaStream to make a copy. If that has no performance ramifications, we can create clones of the mediastream for every peer and this will work.
